File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        dp = [None for i in range(n+1)]
        def f(n, dp):
            if n == 0:
                return 1
            if dp[n]:
                return dp[n]
            ones = f(n-1, dp)
            twos = 0
            if n > 1:
                twos = f(n-2, dp)
            dp[n] = ones + twos
            return dp[n]
        return f(n, dp)
            

File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int f(int ind, vector<int> &dp){
        if(ind == 0){
            return 1;
        }
        if(dp[ind]!=-1){
            return dp[ind];
        }
        int ones = f(ind-1, dp);
        int twos = 0;
        if(ind > 1){
            twos = f(ind-2, dp);
        }
        return dp[ind] = ones + twos;
    }
    int climbStairs(int n) {
        vector<int> dp(n+1, -1);
        return f(n, dp);
    }
};